### PR TITLE
feat : Create compute shared reservation Sample

### PR DESCRIPTION
**Pull request by Thoughtseize1**
## Description

Created Sample for Compute Create Shared Reservation. 

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/client_library/snippets/tests/test_compute_reservation.py [hunk 1]
```diff
@@ -128,3 +124,27 @@ def test_list_compute_reservation(reservation):
...
+        pytest.skip(
+            "Skipping test because SHARED_PROJECT_ID environment variable is not set."
+        )
+    try:
+        response = create_compute_shared_reservation(
+            PROJECT_ID, ZONE, RESERVATION_NAME, SHARED_PROJECT_ID
+        )
+        assert response.share_settings.project_map.values()
+    finally:
+        delete_compute_reservation(PROJECT_ID, ZONE, RESERVATION_NAME)
```
Comment (m-strzelczyk) [inline compute/client_library/snippets/tests/test_compute_reservation.py:150]: use another try and catch to not report an error when the reservation is missing (if creation failed)

**Review by m-strzelczyk — CHANGES_REQUESTED** [review]

File: compute/client_library/ingredients/compute_reservation/create_compute_shared_reservation.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        shared_project_id (str): The ID of the project that the reservation is shared with.
+    Returns:
+        Reservation object that represents the new reservation.
+    """
+
+    instance_properties = compute_v1.AllocationSpecificSKUAllocationReservedInstanceProperties(
+        machine_type="n1-standard-1",
+        # Optional. Specifies amount of local ssd to reserve with each instance.
+        local_ssds=[
+            compute_v1.AllocationSpecificSKUAllocationAllocatedInstancePropertiesReservedDisk(
```
[the remark this answers is not in the document]
Comment (Thoughtseize1) [reply compute/client_library/ingredients/compute_reservation/create_compute_shared_reservation.py:44]: But I don't know what I could do with that 🤷‍♂️
Maybe we could make a request to update the library. Then I could use more readable names here! 😄

File: compute/client_library/snippets/tests/test_compute_reservation.py [hunk 3]
```diff
@@ -128,3 +124,27 @@ def test_list_compute_reservation(reservation):
...
+    by SHARED_PROJECT_ID.
+
+    Make sure to set the SHARED_PROJECT_ID environment variable before running this test,
+    and ensure that the project is allowlisted in the organization policy for shared reservations.
+
+    If the SHARED_PROJECT_ID environment variable is not set, the test will be skipped.
+    """
+    if not SHARED_PROJECT_ID:
+        pytest.skip(
+            "Skipping test because SHARED_PROJECT_ID environment variable is not set."
```
[the remark this answers is not in the document]
Comment (Thoughtseize1) [reply compute/client_library/snippets/tests/test_compute_reservation.py:142]: Done

File: compute/client_library/snippets/tests/test_compute_reservation.py (hunk 1 above)
Comment (Thoughtseize1) [reply compute/client_library/snippets/tests/test_compute_reservation.py]: Added🙌

**Review by m-strzelczyk — APPROVED** [review]

**Pull request merged**